src/datanode/src/server/grpc.rs

```diff
@@ -69,21 +71,41 @@ impl Instance {
     }
 
     pub(crate) async fn handle_alter(&self, expr: AlterExpr) -> Result<Output> {
+        let catalog = expr.catalog_name.clone();
+        let schema = expr.schema_name.clone();
+        let table = expr.table_name.clone();
+
         let request = alter_expr_to_request(expr).context(AlterExprToRequestSnafu)?;
-        self.sql_handler()
+        let ret = self
+            .sql_handler()
             .execute(SqlRequest::Alter(request), QueryContext::arc())
-            .await
+            .await;
+
+        self.invalidate_catalog_cache_if_need(&catalog, &schema, &table)
```

Comment:
Should not call it here, the procedure on metasrv will trigger `invalidate cache`